execution_workflows/Roar/workflow/scripts/make_roar_annotations.py

```diff
@@ -0,0 +1,563 @@
+#!/usr/bin/env python3
+
+import pyranges as pr
+import pandas as pd
+import numpy as np
+import argparse
+import sys
+
+'''
+Script to generate Roar-compliant GTF annotations ('single' APA type currently)
+
+Attempting to recreate 'single' and 'multiple' APA custom annotation files from a GTF of tx & BED file of polyA sites
+
+Instructions are  in the vignette - (https://bioconductor.org/packages/release/bioc/vignettes/roar/inst/doc/roar.pdf)
+
+Example files can be found at the Github repo - (https://github.com/vodkatad/roar/wiki/Identify-differential-APA-usage-from-RNA-seq-alignments#hg38-annotation-files-refseq-version-79)
+
+
+I've used PyRanges (https://github.com/biocore-ntnu/pyranges) throughout for the analysis.
+TL;DR: PyRanges objects are dictionaries of {(chromosome, strand): pd.DataFrame}
+Dataframes store the corresponding intervals.
+pr.apply (/ pr.apply_pair()) is used to expose the underlying dataframes (on which standard pandas functions/methods can be applied)
+
+
+Workflow/interpretation of how to generate these annotations is described below.
+
+**Part 1 - Selecting representative proximal polyA site**
+
+Direct quote from manual:
+
+"Every gene with at least one APA site is considered; their
+longest transcript is used to get the “classical” polyadenylation syte (canonical
+end of the longest transcript stored in UCSC) and the most proximal (with
+respect to the TSS, that is the farther to the canonical end of the transcript;
+but preferring sites still in the 3’UTR) reported APA site found in PolyA DB is
+used to define the end of the short isoform"
+
+1. Overlap GTF with polyA sites, retain gene IDs that have at least 1 overlapping polyA site
+2. Select the longest transcript for each gene
```

Comment:
I completely agree it doesn't make the most sense. I tried to interpret the [documentation](https://www.bioconductor.org/packages/release/bioc/vignettes/roar/inst/doc/roar.pdf) as literally as possible which says the following:

> Every gene with at least one APA site is considered; **their
longest transcript** is used to get the “classical” polyadenylation syte (canonical
end of the longest transcript stored in UCSC) and the most proximal (with
respect to the TSS, that is the farther to the canonical end of the transcript;
but preferring sites still in the 3’UTR) reported APA site found in PolyA DB is
used to define the end of the short isoform

Perhaps it's just an omission and it is meant to be the longest 3'UTR (or the most 'downstream' 3'UTR). I can't see a way to interpret what they've described here differently to how I've implemented it, but then again I've spent a lot of time looking at this so I could well be acting hardheadedly. 

I could get in touch with the author to clarify but think it's unlikely we'll get a response within our timeline (worth a try regardless I suspect). I think the only definitive way to answer this would be if I could regenerate the [provided annotations](https://github.com/vodkatad/roar/wiki/Identify-differential-APA-usage-from-RNA-seq-alignments#hg38-annotation-files-refseq-version-79) and do some comparisons, but I previously had a hard time finding RefSeq v79 GTF...

Thoughts on what I should do next @dominikburri @faricazjj ?

Reply:
Thanks for putting the documentation snippet here :) 
I'm in clinch: should we take it literal and use the longest transcript or interpret it as the longest 3'UTR, just as you describe it.

So how about we keep it as is for now and we contact them? If we hear from them we can still make changes afterwards. I mean if it involves re-running some results, this should be doable (I hope so at least). 
You can also ask if they can give you RefSeq v79 GTF to actually regenerate the annotations you mention. 

What we can also try to do is to simply check in IGV how the resulting GTF looks like for the human or mouse genome and whether the polyA sites roughly overlap with polyA site databases or annotated 3'UTR features... Though not exactly sure what this would gain us.